### ISSUE TITLE
Feature Request: option to include ambient sounds when transmitting

### ISSUE THREAD

**Issue by ArwynFr** [michail-nikolaev/task-force-arma-3-radio#1440]
In order to increase context and immersion, it would be nice if ambient sounds around the player could be transmitted along over the radio. This would make recipients able to hear firefight sounds / other people speaking in the background of the radio transmission when the message is sent under heavy combat situation. That would require radio operators to seek for a quieter location to transmit messages.

Probably not a good idea to make it a mandatory feature.
Sound redirection might be tricky to implement though.

**Comment by Fusselwurm** [comment]
You could probably emulate something like this for selected events?

* add a "gunshot through radio" sound
* give Arma the information which units the player is currently receiving over radio (this info currently resides in Teamspeak only, afaik)
* keep FiredNear event handlers on those units, and have them play the "gunshot through radio" sound

**Comment by kripto202** [comment]
or maybe take the surrounding sounds and have that transmitted through the radio so if someone is shooting next to you, that gun sounds will be transmitted.

**Comment by Fusselwurm** [comment]
@kripto202 no. afaik, all the sound effects etc are done on the *receiving* end. so if I transmitted all my sounds just like that along with my voice, people over the radio *as well as* those in direct voice (standing next to me) would hear the stuff. not good.

**Comment by ImplicitDeny** [comment]
I was thinking about something similar to what @Fusselwurm suggested, maybe with an additionnal sound for explosions. Indeed, gunshots and explosions are the only two sounds that are trully likely to be heard over the voice of the transmitter. And both are detected by FiredNear, so it should probably not be overcomplicated to design this.

The only dark spot is that I have no idea on how is processed the voice, how the radio effect is added to what is audible by the receiver, and so how to add gunshots and explosions over it. Is this a TeamSpeak process ?

**Comment by ArwynFr** [comment]
I can easily do something alike with my sound card because it is made for streaming. It is able to mix game output + microphone into a « virtual in » device that can be used as Teamspeak's input. I have a very superficial understanding of Windows XAudio2 API, but I think this could be achieved through software instead of hardware.

It might introduce some latency, tough.

If I remember correctly, virtual audio sources is a challenging topic, so perhaps it's not worth it.

**Comment by dedmen** [comment]
>people over the radio as well as those in direct voice

That is a good point.. Radio and Direct Voice are the same audio stream. Meaning if someone would talk in direct to you you'd hear the sounds around yourself twice.

**Comment by ImplicitDeny** [comment]
So only adding sound effects for major background sounds is possibly doable, by using Arma-side scripting. This is vert limiting.

**Comment by commy2** [comment]
This is a good idea, and by that I mean this is the worst idea I've ever heard in regards to Arma.
Just close this.

**Comment by Soldia1138** [comment]
Due to the reasons @dedmen stated: It is not possible to handle more than one audio stream. 
Plus: Modern radios do have a very good noise suppression. You won't hear every gunshot on the radios.

**Comment by dedmen** [comment]
TFAR is using throat microphones. There. Problem solved. No ambient sounds. :D
